owl/EASE-OBJ.owl

```diff
@@ -26,6 +26,17 @@
     
 
 
+    <!-- http://www.ease-crc.org/ont/EASE-OBJ.owl#hasShape -->
+
+    <owl:ObjectProperty rdf:about="http://www.ease-crc.org/ont/EASE-OBJ.owl#hasShape">
+        <rdfs:subPropertyOf rdf:resource="http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#hasRegion"/>
+        <rdfs:domain rdf:resource="http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#Entity"/>
```

Comment:
Entity may be too wide a domain for hasShape.

Reply:
I am not sure about being super restrictive here.
What is your suggestion?

First, qualities may refer to regions.
Is this exclusively? i.e. a region may only be the region of some quality concept but not the other? I do not think so.
Take the example of *Kinematics* quality which is kind of a collection of all attributes contributing to kinematics. But these attributes may contribute to other qualities as well.
So I would not be more specific then saying any quality may take this region.

Second, objects may also refer to regions.
i.e. DUL allows to directly relate an Object to some Region. You could say that only PhysicalObjects have a shape, maybe. But what about e.g depictions of social objects?

Last, events may also have regions. Can any event have a shape? Well it could! Like when people are moving around, the bounding box of their movement for example.

Reply:
Likewise any event can have a color. With metaphor you can do anything.

But we're not dealing in metaphors. A shape is a particular kind of thing, and it does make sense to limit it to Physical Objects. A social object cannot have a shape. A physical depiction of it does. I don't see where having shapes for events will prove useful.

Reply:
No it is not a metaphor. The event can have a quality, like the space region in which it took place, which is forming a shape. No metaphor at all

Reply:
"Likewise any event can have a color" -- what about saying that events may have qualities that have color regions? that seems to be fine to me

Reply:
Qualities definitely must be in the domain of any property with range Region, so there's no objection there.

Reply:
But qualities are generally optional in DUL. So you can relate Event/Object directly to regions without going via Quality. Hence, I used Entity as domain

Reply:
The more things we allow the less the ontology can tell us; for example, if X hasShape something, you know nothing of X if the domain of hasShape is Entity.

Perhaps this is sometimes appropriate, but for an ontology of everyday activity it seems plausible that we can consider only physical objects, and their appearances, to have shapes.

Reply:
But what does "X is a quality" buy us? Or it could be "X is a (physical) object or quality", also does not seem to be helpful.

More useful are axioms at the object concepts, that restrict their qualities, I think. But the property domain of hasRegion sub-properties does not seem to be super relevant for reasoning, or am I missing something?

Reply:
I would say this is a general principle, to try and think of as many restrictions as will not hurt the class of models, and put them in. It is easy to not specify a domain for a property, but I think in our case we can specify a domain without missing out on any useful models for our application, and therefore we should.

Reply:
So what is your suggestion?

Reply:
PhysicalObject and Appearance (the corresponding quality of shape).

Reply:
I am trying to follow principles of DUL here. And as explained before, DUL allows to put regions on objects and events. I *was* thinking about this before, and decided that this would be in the sense of DUL

Reply:
> PhysicalObject and Appearance (the corresponding quality of shape).

I am still questioning if there is one **corresponding** quality. I was arguing before that attributes may contribute to multiple qualities. And the shape may not only contribute to appearance

Reply:
"Contribute" is also vague. Mass is a quality; it should also correspond to a region in positive real number space, or some qualitative space on top of such. Many things can contribute to Mass, but it would not be appropriate to say that Mass takes values in a region for material density. That's not what Mass is.

Reply:
Mass is not a quality in DUL, it is a *PhysicalAttribute*

Reply:
I assume you know what I meant with contribute. that should be obvious: i.e. that it's a region of that quality.

Reply:
No actually I don't, since we're debating what I thought obvious, that only Physical Objects and the appearance quality should have a shape.

Reply:
ok sorry for being unclear

Reply:
So regarding that, we would need to clarify if there is a 1-to-1 or 1-to-many mapping between Regions and Qualities. So the question can a region be the region of two distinct qualities?
It is hard to answer this question without having a complete taxonomy of qualities in DUL. Actually that makes it hard to think about what are all the qualities that are out there

Reply:
So thinking of Qualities as *qualifications* of quantitative regions, I think it makes sense that the same region can be qualified in different ways, generally